src/components/BulkEditList/BulkEditListResult/BulkEditInApp/ContentUpdatesForm/ContentUpdatesForm.js

```diff
@@ -191,30 +162,25 @@ export const ContentUpdatesForm = ({
     onContentUpdatesChanged(mappedContentUpdates);
   }, [fields]);
 
-  const renderOptions = (array) => {
-    return array.map(item => {
-      if (typeof item === 'object' && Object.keys(item).length === 1) {
-        const category = Object.keys(item)[0];
-        let categoryLabel = '';
-        const categoryOptions = item[category].map(option => {
-          categoryLabel = option.categoryLabel;
+  const renderOptions = (optionsMap) => {
+    const result = [];
 
-          return (
-            <option key={option.value} value={option.value} disabled={option.disabled}>{option.label}</option>
-          );
-        });
-
-        return (
+    for (const [category, item] of Object.entries(optionsMap)) {
+      if (Array.isArray(item)) {
+        result.push((
           <optgroup
             key={category}
-            label={categoryLabel}
+            label={category}
           >
-            {categoryOptions}
-          </optgroup>);
+            {renderOptions(item)}
+          </optgroup>
+        ));
       } else {
-        return <option key={item.value} value={item.value} disabled={item.disabled}>{item.label}</option>;
+        result.push((<option key={item.value} value={item.value} disabled={item.disabled}>{item.label}</option>));
       }
-    });
+    }
+
+    return result.map(item => item);
```

Comment:
What is the reason to use `.map` here?

Reply:
I would replace whole `for (const [category, item] of Object.entries(optionsMap)) {` by this map

Reply:
@mariia-aloshyna  and @NikitaSedyx I rewrited by single `map`. Thanks for advice

Reply:
Thank you!